x-pack/plugins/apm/server/saved_objects/migrations/migrate_legacy_apm_indices_to_space_aware.ts

```diff
@@ -57,7 +57,6 @@ export async function migrateLegacyAPMIndicesToSpaceAware({
       type: 'space',
       page: 1,
       perPage: 10_000, // max number of spaces as of 8.2
-      fields: ['name'], // to avoid fetching *all* fields
```

Comment:
This seems like a pretty big change in terms of performance. Is the suggestion to avoid using SOs for documents with many fields (or fields containing big blobs of data) where only few or lightweight fields are needed in the majories of queries?

We are already using custom (system) indices for certain features (agent configs, custom links). This seems like yet another reason to avoid SOs.